merlin/dag/node.py

```diff
@@ -663,15 +671,18 @@ def _filter_by_type(elements, type_):
     return results
 
 
-def _combine_schemas(elements):
+def _combine_schemas(elements, input_schemas=False):
```

Comment:
Seems like one approach could be to implement methods like this in order to avoid the boolean flag:
* `_combine_input_schemas(elements)
* `_combine_output_schemas(elements)
* `_combine_schemas(schemas: List[Schema])